pkg/cmd/run/list/list.go

```diff
@@ -108,6 +108,10 @@ func listRun(opts *ListOptions) error {
 		return fmt.Errorf("failed to get runs: %w", err)
 	}
 
+	if len(runs) == 0 {
+		return utils.HandleNoResults(opts.IO, "No runs found")
```

Comment:
Despite comments, this works correctly when a repo has no runs. Additionally, we should delete lines 129-134.

Reply:
Are there any other places with no longer necessary empty list handling?

Reply:
I think I hit most of it, but I'll make another pass through

Reply:
Ooh, `gh pr checks` is a really annoying case. Cause the null check is too deep, so it'd require some refactoring.

One option would be to turn `NoResultsError` into an _actual_ error which is returned here, and then suppress the error and return `ExitOK` in `main.go`, by adding to the block:
```golang
	if cmd, err := rootCmd.ExecuteC(); err != nil {
		var pagerPipeError *iostreams.ErrClosedPagerPipe
		if err == cmdutil.SilentError {
			return exitError
		} else if cmdutil.IsUserCancellation(err) {
			if errors.Is(err, terminal.InterruptErr) {
				// ensure the next shell prompt will start on its own line
				fmt.Fprint(stderr, "\n")
			}
			return exitCancel
		} else if errors.Is(err, authError) {
			return exitAuth
		} else if errors.As(err, &pagerPipeError) {
			// ignore the error raised when piping to a closed pager
			return exitOK
		} else if errors.As(err, &noResultsError) {
		        // print the error message to stderr if cmdFactory.IOStreams.IsStdoutTTY(), else silence
                        return exitOK
		}
```

Reply:
I actually much prefer this approach of capturing the `NoResultsError` here in the main block as then returning `exitOK`. I think it is cleaner and allows the use of `cmdutil.NoResultError` without encapsulating the `io.IsStdoutTTY()` logic in the error definition. Would it be asking to much to refactor this PR to use this new approach?

Reply:
not at all, i think i have that diff lying around already